File: test-endpoints.py
#!/usr/bin/env python3
import os
import json
import requests
from datetime import datetime

# Configurações
BACKEND_URL = 'http://localhost:3000/api'
FRONTEND_URL = 'http://localhost:3001'
OUTPUT_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'endpoint-tests')
TEST_USER = {
    'email': 'user1@devagram.com',
    'password': '2700'
}

# Criar diretório de saída se não existir
os.makedirs(OUTPUT_DIR, exist_ok=True)

# Função para salvar resposta em arquivo
def save_response(filename, data):
    file_path = os.path.join(OUTPUT_DIR, filename)
    
    if isinstance(data, requests.Response):
        try:
            content = data.json()
            with open(file_path, 'w', encoding='utf-8') as f:
                json.dump({
                    'status': data.status_code,
                    'headers': dict(data.headers),
                    'data': content
                }, f, indent=2)
        except:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(data.text)
    else:
        with open(file_path, 'w', encoding='utf-8') as f:
            if isinstance(data, dict):
                json.dump(data, f, indent=2)
            else:
                f.write(str(data))
    
    print(f"Resposta salva em {file_path}")

# Função para testar um endpoint
def test_endpoint(name, method, url, headers=None, data=None):
    print(f"\n> Testando endpoint: {name}")
    print(f"  {method} {url}")
    
    if headers is None:
        headers = {}
    
    try:
        response = requests.request(
            method=method,
            url=url,
            headers=headers,
            json=data if data and method != 'GET' else None,
            params=data if data and method == 'GET' else None
        )
        
        filename = f"{name.lower().replace(' ', '_')}.json"
        save_response(filename, response)
        
        if 200 <= response.status_code < 300:
            print(f"  Resultado: OK ({response.status_code})")
        else:
            print(f"  Resultado: ERRO ({response.status_code})")
        
        return response
    except Exception as e:
        print(f"  Erro ao testar endpoint: {str(e)}")
        return None

# Função principal para executar todos os testes
def run_tests():
    print('===== Testando endpoints do Devagram =====')
    token = None
    
    # 1. Testar login e obter token
    print('\n=== Testando autenticação ===')
    login_response = test_endpoint(
        'Login',
        'POST',
        f"{BACKEND_URL}/login",
        {'Content-Type': 'application/json'},
        {'login': TEST_USER['email'], 'senha': TEST_USER['password']}
    )
    
    if login_response and login_response.status_code == 200:
        try:
            token = login_response.json().get('token')
            if token:
                print('  Token obtido com sucesso')
            else:
                print('  Token não encontrado na resposta')
        except:
            print('  Falha ao processar resposta de login')
    else:
        print('  Falha ao fazer login')
    
    # 2. Testar endpoints públicos
    print('\n=== Testando endpoints públicos ===')
    
    test_endpoint(
        'Frontend Home',
        'GET',
        FRONTEND_URL
    )
    
    test_endpoint(
        'Frontend Reset',
        'GET',
        f"{FRONTEND_URL}/reset.html"
    )
    
    # 3. Testar endpoints autenticados
    if token:
        print('\n=== Testando endpoints autenticados ===')
        auth_headers = {'Authorization': f'Bearer {token}'}
        
        # Feed
        test_endpoint(
            'Feed',
            'GET',
            f"{BACKEND_URL}/feed",
            auth_headers
        )
        
        # Pesquisa
        test_endpoint(
            'Pesquisa',
            'GET',
            f"{BACKEND_URL}/pesquisa",
            auth_headers,
            {'filtro': 'user'}
        )
        
        # Usuário
        test_endpoint(
            'Usuario',
            'GET',
            f"{BACKEND_URL}/usuario",
            auth_headers
        )
        
        # Seguidor
        test_endpoint(
            'Seguidor',
            'GET',
            f"{BACKEND_URL}/seguidor",
            auth_headers
        )
        
        # Comentário (apenas verificação)
        test_endpoint(
            'Comentario Options',
            'OPTIONS',
            f"{BACKEND_URL}/comentario",
            auth_headers
        )
        
        # Like (apenas verificação)
        test_endpoint(
            'Like Options',
            'OPTIONS',
            f"{BACKEND_URL}/like",
            auth_headers
        )
    
    print('\n=== Testes concluídos ===')
    print(f"Resultados salvos na pasta: {OUTPUT_DIR}")
    print(f"Data e hora dos testes: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")

if __name__ == "__main__":
    run_tests()

File: test-endpoints.js
const axios = require('axios');
const fs = require('fs');
const path = require('path');

// Configurações
const BACKEND_URL = 'http://localhost:3000/api';
const FRONTEND_URL = 'http://localhost:3001';
const OUTPUT_DIR = path.join(__dirname, 'endpoint-tests');
const TEST_USER = {
  email: 'user1@devagram.com',
  password: '2700'
};

// Criar diretório de saída se não existir
if (!fs.existsSync(OUTPUT_DIR)) {
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });
}

// Função para salvar resposta em arquivo
const saveResponse = (filename, data) => {
  const filePath = path.join(OUTPUT_DIR, filename);
  fs.writeFileSync(filePath, JSON.stringify(data, null, 2));
  console.log(`Resposta salva em ${filePath}`);
};

// Função para testar um endpoint
const testEndpoint = async (name, method, url, headers = {}, data = null) => {
  console.log(`\n> Testando endpoint: ${name}`);
  console.log(`  ${method} ${url}`);
  
  try {
    const config = {
      method,
      url,
      headers,
      validateStatus: () => true // Não rejeitar em caso de erro HTTP
    };
    
    if (data) {
      config.data = data;
    }
    
    const response = await axios(config);
    
    const filename = `${name.toLowerCase().replace(/\s+/g, '_')}.json`;
    saveResponse(filename, {
      status: response.status,
      statusText: response.statusText,
      data: response.data
    });
    
    if (response.status >= 200 && response.status < 300) {
      console.log(`  Resultado: OK (${response.status})`);
    } else {
      console.log(`  Resultado: ERRO (${response.status})`);
    }
    
    return response;
  } catch (error) {
    console.error(`  Erro ao testar endpoint: ${error.message}`);
    return null;
  }
};

// Função principal para executar todos os testes
const runTests = async () => {
  console.log('===== Testando endpoints do Devagram =====');
  let token = null;
  
  // 1. Testar login e obter token
  console.log('\n=== Testando autenticação ===');
  const loginResponse = await testEndpoint(
    'Login',
    'POST',
    `${BACKEND_URL}/login`,
    { 'Content-Type': 'application/json' },
    { login: TEST_USER.email, senha: TEST_USER.password }
  );
  
  if (loginResponse && loginResponse.data && loginResponse.data.token) {
    token = loginResponse.data.token;
    console.log('  Token obtido com sucesso');
  } else {
    console.log('  Falha ao obter token');
  }
  
  // 2. Testar endpoints públicos
  console.log('\n=== Testando endpoints públicos ===');
  
  await testEndpoint(
    'Frontend Home',
    'GET',
    FRONTEND_URL
  );
  
  await testEndpoint(
    'Frontend Reset',
    'GET',
    `${FRONTEND_URL}/reset.html`
  );
  
  // 3. Testar endpoints autenticados
  if (token) {
    console.log('\n=== Testando endpoints autenticados ===');
    const authHeaders = { 'Authorization': `Bearer ${token}` };
    
    // Feed
    await testEndpoint(
      'Feed',
      'GET',
      `${BACKEND_URL}/feed`,
      authHeaders
    );
    
    // Pesquisa
    await testEndpoint(
      'Pesquisa',
      'GET',
      `${BACKEND_URL}/pesquisa?filtro=user`,
      authHeaders
    );
    
    // Usuário
    await testEndpoint(
      'Usuario',
      'GET',
      `${BACKEND_URL}/usuario`,
      authHeaders
    );
    
    // Seguidor
    await testEndpoint(
      'Seguidor',
      'GET',
      `${BACKEND_URL}/seguidor`,
      authHeaders
    );
    
    // Comentário (apenas verificação)
    await testEndpoint(
      'Comentario Options',
      'OPTIONS',
      `${BACKEND_URL}/comentario`,
      authHeaders
    );
    
    // Like (apenas verificação)
    await testEndpoint(
      'Like Options',
      'OPTIONS',
      `${BACKEND_URL}/like`,
      authHeaders
    );
    
    // Publicação (apenas verificação)
    await testEndpoint(
      'Publicacao Options',
      'OPTIONS',
      `${BACKEND_URL}/publicacao`,
      authHeaders
    );
  }
  
  console.log('\n=== Testes concluídos ===');
  console.log(`Resultados salvos na pasta: ${OUTPUT_DIR}`);
};

// Executar testes
runTests().catch(error => {
  console.error('Erro ao executar testes:', error);
});